daemon/src/maker.rs

```diff
@@ -204,6 +195,31 @@ async fn main() -> Result<()> {
                 });
 
                 tokio::spawn(maker_inc_connections_address.attach_stream(listener_stream));
+
+                // XXX: Is there a way to not repeat this in both maker and taker?
```

Comment:
I would not bother too much, duplicate for now. As long as it is gluecode that's fine. 

Please remove the `XXX` comments before merging - or convert them to `TODO`. I would avoid having multiple different types of actionable comments in there. Will make it hard to find them.

Reply:
I can leave it as-is, the question was more to find out what sort of Rust wizardry I still have to learn :D 

I typically put `XXX` when I ask questions that are meant for the review, but which should stay inside when merged. So, don't worry, I'm onboard with sticking to `TODO`s in the codebase 💯  happy to use `TODO` for that as well, but I didn't want to accidentally push some transient comments in.

Reply:
thank you @thomaseizinger ! I'll give it a shot, but as a separate cleanup PR, to not to drag this one.